### PR TITLE
Wizard: Config validation button for Icinga 2 API

### LINKED ISSUE #3101: Wizard: Config validation button for Icinga 2 API
When the transport type is `Icinga 2 API`

The `Validate Config` button just refreshes the form.

On `Next` config is actually validated.

### PR DESCRIPTION
**Pull request by lazyfrosch**
This fixes 2 issues:

- [x] Validation and Skip Validation was not handled correctly
- [x] How the Validation button is displayed in Wizard and Form

@lippserd please have a look, especially on the TODOs left

fixes #3101

### CODE REVIEW

**Comment by lippserd** [comment]
@nilmerg Please have a look at this one.

File: modules/monitoring/application/forms/Config/TransportConfigForm.php [hunk 1]
```diff
@@ -270,46 +270,49 @@ public function addSubmitButton()
         parent::addSubmitButton();
 
         if ($this->getSubForm('transport_form') instanceof ApiTransportForm) {
```
Comment (nilmerg) [inline modules/monitoring/application/forms/Config/TransportConfigForm.php]: Please relax this check as in the review summary stated.

File: modules/monitoring/application/forms/Config/TransportConfigForm.php [hunk 2]
```diff
@@ -351,9 +354,12 @@ public function isValid($formData)
...
-            $this->isSubmitted() && isset($formData['force_creation']) && $formData['force_creation'])
-        )) {
+        if ($this->getSubForm('transport_form') instanceof ApiTransportForm) {
+            if (isset($formData['force_creation']) && $formData['force_creation']) {
+                // ignore any validation result
+                return true;
+            }
+
             try {
                 CommandTransport::createTransport(new ConfigObject($this->getValues()))->probe();
```
Comment (nilmerg) [inline modules/monitoring/application/forms/Config/TransportConfigForm.php]: Please only do this if the transport has such a method. If not, just return true as if the test succeeded.

File: modules/monitoring/application/forms/Config/TransportConfigForm.php [hunk 3]
```diff
@@ -351,9 +354,12 @@ public function isValid($formData)
             return false;
         }
 
-        if (! ($this->getElement('transport_validation') === null || (
-            $this->isSubmitted() && isset($formData['force_creation']) && $formData['force_creation'])
-        )) {
+        if ($this->getSubForm('transport_form') instanceof ApiTransportForm) {
```
[the remark this answers is not in the document]
Comment (lazyfrosch) [reply modules/monitoring/application/forms/Config/TransportConfigForm.php]: You weren't able to skip validation before this change, so I had to fix it...

File: modules/monitoring/application/forms/Config/TransportConfigForm.php [hunk 4]
```diff
@@ -366,10 +372,10 @@ public function isValid($formData)
                     'checkbox',
                     'force_creation',
                     array(
-                        'order'         => 0,
```
[the remark this answers is not in the document]
Comment (lazyfrosch) [reply modules/monitoring/application/forms/Config/TransportConfigForm.php]: I only fixed indentation here...

**Comment by lazyfrosch** [comment]
Please to required changes on your own, I'm busy with other things today.

And I guess you want to fix this for 2.5.0, so it works... Details can be done later...

**Pull request merged**